src/main/java/org/jitsi/videobridge/IceUdpTransportManager.java

```diff
@@ -391,65 +396,67 @@ public boolean addChannel(Channel channel)
         if (closed)
             return false;
 
-        if (channel instanceof SctpConnection
-                && sctpConnection != null
-                && sctpConnection != channel)
-        {
-            logd("Not adding a second SctpConnection to TransportManager.");
-            return false;
-        }
-
-        if (!super.addChannel(channel))
-            return false;
-
-        if (channel instanceof SctpConnection)
+        synchronized (channelForDtlsLock)
         {
-            // When an SctpConnection is added, it automatically replaces
-            // channelForDtls, because it needs DTLS packets for the application
-            // data inside them.
-            sctpConnection = (SctpConnection) channel;
-            if (channelForDtls != null)
+            if (channel instanceof SctpConnection
+                    && sctpConnection != null
+                    && sctpConnection != channel)
             {
-                /*
-                 * channelForDtls is necessarily an RtpChannel, because we don't
-                 * add more than one SctpConnection. The SctpConnection socket
-                 * will automatically accept DTLS.
-                 */
-                RtpChannel rtpChannelForDtls = (RtpChannel) channelForDtls;
```

Comment:
This can still cause a cast exception.  I can submit another pull request with the fix.  The comment is misleading as this can change types during a close.